### PR TITLE
Adding ScrollDetector

### PR DESCRIPTION
**Pull request by hastebrot**
# Description

This pull request is based on #461 and adds `ScrollDetector`. This is useful for detecting mouse scroll wheel inputs or touchpad inputs.

~Creating as a draft since this is based on the `erick.mouse-hover` which in turn is based on `develop`. The draft state can be changed once #461 was merged.~

Screen Recording 2020-09-11 at 19 35 24 mov

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [ ] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [ ] The continuous integration (CI) is passing

### CODE REVIEW

**Comment by erickzanardo** [comment]
@hastebrot #461 has been merged, please rebase your branch with develop so we can review this

**Comment by hastebrot** [comment]
> please rebase your branch with develop so we can review this

I've rebased my branch with `develop`.

File: lib/game/widget_builder.dart [hunk 1]
```diff
@@ -208,7 +209,12 @@ Widget _applyBasicGesturesDetectors(Game game, Widget child) {
 
 Widget _applyMouseDetectors(game, Widget child) {
   return MouseRegion(
-    child: child,
+    child: Listener(
+        child: child,
+        onPointerSignal: (event) =>
+            game is ScrollDetector && event is PointerScrollEvent
+                ? game.onScroll(event)
+                : null),
```
Comment (erickzanardo) [inline lib/game/widget_builder.dart:217]: Add a trialing comma here please. Like:

```javascript
    : null,
),
```

File: lib/gestures.dart [hunk 2]
```diff
@@ -103,3 +103,7 @@ mixin ScaleDetector on Game {
 mixin MouseMovementDetector on Game {
   void onMouseMove(PointerHoverEvent event) {}
 }
+
+mixin ScrollDetector on Game {
```
Comment (erickzanardo) [inline lib/gestures.dart:107]: I think we should name this `MouseScrollDetector`, What do you think?

**Review by erickzanardo — APPROVED** [review]
I left a couple of suggestions, but this looks good to me already.

File: lib/gestures.dart (hunk 2 above)
Comment (luanpotter) [reply lib/gestures.dart:107]: Assuming it only triggers via mouse, and not mobile touch scroll, yes.

**Review by luanpotter — APPROVED** [review]
I love this! I was actually missing this feature in a game I was making. Awesome!

File: lib/gestures.dart (hunk 2 above)
Comment (erickzanardo) [reply lib/gestures.dart:107]: @hastebrot can you confirm this? So we can merge this?

**Review by spydon — APPROVED** [review]

**Pull request merged**